extensions/skins/conversation_v1/Conversation.js

```diff
@@ -589,7 +588,7 @@ oppia.directive('conversationSkin', [function() {
       });
 
       $window.addEventListener('beforeunload', function(e) {
-        if (hasInteractedAtLeastOnce && !$scope.explorationCompleted &&
+        if (hasInteractedAtLeastOnce && !$scope.isOnFinalCard() &&
```

Comment:
I'm not sure if I completely understand. Do you mean something like oppiaPlayerService.isStateTerminal($scope.transcript[$scope.transcript.length - 1])?